### DISCUSSION TITLE
Steam release of OpenRCT2

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#18367] [Feature requests]
Similar to our sister project [OpenTTD](https://github.com/OpenTTD/OpenTTD), I think it's be great for expanding our community to release the game to steam.
I have done quite a bit of research into it and it should be doable right now to get the game up and running on steam.
Let me sum up why we should do this and what the advantages are.

* Automatic distribution of updates to the entire community.
* The ability to automate updates in daily, weekly or commit based updates as "Beta" releases (configurable in your steam library).
* The ability to add achievements
* People can track their time playing the game and overall more statistics.
* Exposure to a massive community of gamers that might not know of the project.
* **Access to the Steam Workshop** where asset packs, objects, plugins, scenarios, custom track and plugins can be uploaded by the community, push updates to which will be downloaded automatically and other things. (It'd still work without steam aswell ofcourse but it'd be one united hub where people can upload their RCT things, and OpenRCT2 could pull missing objects and whatnot from).
* Cloud save backup
* Steam Input profile sharing (particularly useful for Steam Deck)
* Grid art
* Community Hub

### What does it take?
The cost of a steam release is a one time fee of 100 USD (and 30% of all profits, but since the game is free, that doesn't exist).  We can do this right now and set it up as early access until the game isn't reliant on RCT2 & RCTC anymore.

We should link to RCT2 and RCTC in the description of early access to keep Atari happy as well. 

OpenTTD has their Betas set up as following:
![image](https://user-images.githubusercontent.com/1516893/196254833-4a4d83cf-b4c4-4c23-b258-051f8b099979.png)

### Steam Workshop integration
The current issue with our community is that it's scattered everywhere. Some things are in discord, plugins are on a dedicated website, maps are on NEDesigns and RCTGo amongst others. Objects are everywhere and nowhere, on now dissolved websites and lost for the end of time, etc etc etc.
To unite all of these (without excluding any old techniques) the Steam Workshop will be the perfect place. Games like Planet Coaster and Cities: Skylines us it to their advantage to ship user created content to a wide public without the chance of things getting lost forever due to a small community site ceasing to exist.
The steam work is sorted out well and it's extremely easy to use, just click subscribe and next time on loading it will be available in your game.
![image](https://user-images.githubusercontent.com/1516893/196256229-4524ca93-6b4b-4b80-9dba-f2990924026b.png)
It's a perfect community tool for seeing what is the new and flashy product. People can collaborate on items and update them together, etc.

The tag system can make differences between the main and subtags.
![image](https://user-images.githubusercontent.com/1516893/196256901-6b181ffb-0225-4afc-a3d0-6026322cb24e.png)

The nice thing is that people could even make it so things are dependent on the other. Useful if someone has made a vehicle for a track type that doesn't exist in vanilla OpenRCT2, once those objects become a thing.

Tags with their subtags for OpenRCT2 would be:
* Assetpack
  *  Sound Effects replacement
  *  Graphics replacement
* Plugin
* Scenario
* Park
* Object
  * Ride
    * Roller Coaster
    * Water ride
    * Gentle ride
    * Thrill ride
    * Transport ride
    * Stall
  * Music
  * Path
    * Railings
    * Queues
    * Surfaces
  * Scenery items
    * Large scenery
    * Small scenery
    * Path objects
    * Banners
    * Walls and fences
  * Station styles
  * Park entrances
  * Palettes/Water
  * Scenery groups
  * Terrain
    * Surface
    * Edge

Please let me know if you wish to know more specific things about this.

**Comment by James103** [comment]
> * Access to the Steam Workshop

As far as I can tell, this requires integration with the Steam SDK, whose license may not be compatible with OpenRCT2's GPL-3 license.

Steam SDK access agreement: https://partner.steamgames.com/documentation/sdk_access_agreement
OpenRCT2's license: https://github.com/OpenRCT2/OpenRCT2/blob/develop/licence.txt (GPL-3)

**Reply by IntelOrca** [reply]
OpenTTD is also GPL

**Reply by tsunamistate** [reply]
OpenTTD doesn't use the Workshop, it uses in-game content manager instead

**Reply by James103** [reply]
From glx22 on OpenTTD's official Discord:

> OpenTTD doesn't use steam integration, it's not compatible with GPL

Link to message: https://discord.com/channels/142724111502802944/1008473180924543128/1031707180463833159

**Comment by hmmmmmmm** [comment]
How does this square with the necessity to own the original game (for the time being, at least)?

**Reply by karst** [reply]
The game will not launch without RCT2 or RCTC loaded in as base game during early access, just like how it works like right now.

**Reply by hmmmmmmm** [reply]
What I mean is, would Steam agree to distribute a game that doesn't work without relying on data from another?

**Reply by James103** [reply]
The game currently requires assets from RCT2, RCT1, and/or RCT Classic to run, all of which are copyrighted by third parties who are unlikely to accept a free release of the game on Steam.

The best solution for now (IMO) is to wait until version 1.0.0 is released before the game is available on Steam.
Version 1.0.0, in turn, requires OpenGraphics (OpenGFX), OpenSFX, and OpenMSX for OpenRCT2 to be completed and ready for production use, as well as a default set of scenarios for both new and established players.

**Reply by karst** [reply]
Steam literally releases paid mods for their own games. They are released like games, but act like DLC (as you need the original game for it to function). This is how CS:GO started as well as Half-Life. Portal has mods on the steam store, etc. So I doubt that it will be denied because of it. Especially as long as it's in early access modus. Not to mention that another platform GOG/CD Projekt has asked before if they could bundle OpenRCT2 with RCT2TT on their platform, so I am quite confident it should be fine.

**Reply by hmmmmmmm** [reply]
I think there is a difference between mods which are always intended to rely on the base game, and OpenRCT2/other releases which initially rely on a base game for assets but intend to move away from them in the future. On Steam, mods can be linked to their base games but will always be listed as mods. As far as I am aware, there is no way to "unlink" a mod from its game when the base game becomes unnecessary, requiring a new Steam listing.

**Reply by karst** [reply]
The ability to load in RCT2 and RCTC will always stay, not needing them as base game is just an addition (pretty big one at that but still, doesn't change much)

**Reply by karst** [reply]
> The game currently requires assets from RCT2, RCT1, and/or RCT Classic to run, all of which are copyrighted by third parties who are unlikely to accept a free release of the game on Steam.

They do not get shipped with OpenRCT2, you require the purchase the game, driving up sales for Atari aside from things. I doubt this will be an issue, there is no copyright infringement.

**Reply by hmmmmmmm** [reply]
The issue I see is, if OpenRCT2 is shipped as a game right now, you would not be able to launch the game without the base assets. That is where Steam would have an issue, regardless of where those assets are from. Game (not mod) releases are required to work with no additional work, and releasing as a mod would require creating a new listing at that point. I think it's simpler and less risky to wait until the open replacements are in place before a Steam release.

**Reply by karst** [reply]
The game does launch. It prompts you with a need to install a base game, but it still launches (it doesn't crash). This community very badly needs something like the steam workshop. Waiting for OpenGraphics to be completed could take years. I can see it speeding up once Graphics assetpacks become a thing, but it's gonna be a while. If it's possible I personally would prefer it sooner rather than later.

**Reply by hmmmmmmm** [reply]
Okay, yes, it launches but it is fundamentally unplayable without the original assets. That is the central issue. I don't think it would be accepted by Steam, even with a note in the store page telling people they need one of the original games.

**Reply by IntelOrca** [reply]
Do Valve check new games now days? I have heard there are many entries added to Steam that do nothing more than award you with a bunch of achivements and have no gameplay. There are also tools, emulators and various other things that don't even qualify as games, e.g. RetroArch. RetroArch being an example of something that is intended to play game content sourced by the player.

**Comment by ocalhoun6** [comment]
Putting it on Steam would put us in more direct competition with Classic, which is also on Steam, right? Of course, not *really* since you still need the base game anyway, but the owners of Classic might see it that way. Which means there's a slight possibility that this might bring legal heat?

**Comment by ZeeMaji** [comment]
I don't really like the idea of workshop support, as i feel it would lock a lot of user content to the steam version of openrct2 due to how the workshop works. Along with said user content becoming a lot more vulnerable to being lost due to how cumbersome it is to archive steam workshop content.
I think it would be much more beneficial to the wider community if we had our own independent content repository similar to what openttd has.

**Reply by karst** [reply]
I think there's solutions for that. Cities Skylines is available on many PC platforms but all can still get the content from the Steam Workshop. I do not know the possibilities, but you could for example also make the game auto pull missing files from the workshop. I am not saying to gatekeep most items for people to be forced to use the Steam release of the game. Perhaps it could even backup a database to another website, openrct2.org even maybe (if that gets picked up by the devteam rather than being third party), so people can download from there as well.

**Comment by IntelOrca** [comment]
Three more benefits to add to your list:
* Cloud saves
* Sharing Steam Input profiles (particularly useful for Steam Deck).
* Official grid art, without needing to import custom art from https://www.steamgriddb.com/game/5248315.

**Reply by karst** [reply]
Added them to the list :)

**Comment by karst** [comment]
Auto pushed updates for OpenTTD is fixed here https://github.com/OpenTTD/OpenTTD/blob/master/.github/workflows/release.yml

See msg in OpenTTD discord: https://discord.com/channels/142724111502802944/1008473180924543128/1031743065913839717
